src/Service/SearchService.php

```diff
@@ -205,15 +205,14 @@ public function getDefaultSearch(array $contentTypes = []): Search
             }
         }
 
-        if (!$search instanceof Search) {
-            $search = $searchRepository->findOneBy([
-                'default' => true,
-            ]);
-            if ($search instanceof Search and \count($search->getContentTypes()) > 0) {
-                $contentTypesNotCovertByTheDefaultSearch = \array_diff($contentTypes, $search->getContentTypes());
-                if (\count($contentTypesNotCovertByTheDefaultSearch) > 0) {
-                    $search = null;
-                }
+        /** @var Search|null $search */
+        $search = $searchRepository->findOneBy([
```

Comment:
This seems a breaking change, as $search is set potentially at line 204 and now you are always overriding it!